oak-search-elastic/src/test/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticConnectionRule.java

```diff
@@ -83,33 +128,62 @@ public ElasticConnection getElasticConnectionFromString() {
             int port = uri.getPort();
             String query = uri.getQuery();
 
-            String api_key = null;
-            String api_secret = null;
+            String apiKey = null;
+            String apiSecret = null;
             if (query != null) {
-                api_key = query.split(",")[0].split("=")[1];
-                api_secret = query.split(",")[1].split("=")[1];
+                apiKey = query.split(",")[0].split("=")[1];
+                apiSecret = query.split(",")[1].split("=")[1];
             }
             return ElasticConnection.newBuilder()
                     .withIndexPrefix(indexPrefix + System.currentTimeMillis())
                     .withConnectionParameters(scheme, host, port)
-                    .withApiKeys(api_key, api_secret)
+                    .withApiKeys(apiKey, apiSecret)
                     .build();
         } catch (URISyntaxException e) {
+            LOG.error("Provided elastic connection string  is not valid ", e);
             return null;
         }
     }
 
+    private boolean isValidUri(String connectionString) {
+        if (connectionString == null) {
+            return false;
+        }
+        try {
+            new URI(connectionString);
+            return true;
+        } catch (URISyntaxException e) {
+            LOG.debug("Provided elastic connection string  is not valid ", e);
```

Comment:
extra space. Could it be useful to print `connectionString` as part of the log message?
```suggestion
            LOG.debug("Provided elastic connection string is not valid ", e);
```

Reply:
Exception, logs the malformed uri. So didn't logged it again in message.